### ISSUE TITLE
Http2ServerSpec: The Http/2 server implementation should support multiple concurrent substreams should close substreams when connection is shutting down

### ISSUE THREAD

**Issue by raboof** [akka/akka-http#2882]
https://jenkins.akka.io:8498/job/akka-http-nightly/AKKA_VERSION=default,Node=akka-http-nightly-node,SCALA_VERSION=2.12.10,jdk=AdoptOpenJDK%2011/755/testReport/junit/akka.http.impl.engine.http2/Http2ServerSpec/The_Http_2_server_implementation_should_support_multiple_concurrent_substreams_should_close_substreams_when_connection_is_shutting_down/

```
sbt.ForkMain$ForkError: org.scalatest.exceptions.TestFailedException: WINDOW_UPDATE was not equal to DATA
	at org.scalatest.MatchersHelper$.indicateFailure(MatchersHelper.scala:343)
	at org.scalatest.Matchers$AnyShouldWrapper.shouldBe(Matchers.scala:6919)
	at akka.http.impl.engine.http2.Http2FrameProbe$$anon$1.expectFrameFlagsStreamIdAndPayload(Http2FrameProbe.scala:172)
	at akka.http.impl.engine.http2.Http2FrameProbe$$anon$1.expectFrameFlagsAndPayload(Http2FrameProbe.scala:166)
	at akka.http.impl.engine.http2.Http2FrameProbe$$anon$1.expectDATAFrame(Http2FrameProbe.scala:105)
	at akka.http.impl.engine.http2.Http2FrameProbe$$anon$1.collectMore$1(Http2FrameProbe.scala:113)
	at akka.http.impl.engine.http2.Http2FrameProbe$$anon$1.expectDATA(Http2FrameProbe.scala:125)
	at akka.http.impl.engine.http2.Http2FrameProbe$$anon$1.expectDATA(Http2FrameProbe.scala:129)
	at akka.http.impl.engine.http2.Http2FrameProbeDelegator.expectDATA(Http2FrameProbe.scala:72)
	at akka.http.impl.engine.http2.Http2FrameProbeDelegator.expectDATA$(Http2FrameProbe.scala:72)
	at akka.http.impl.engine.http2.Http2ServerSpec$TestSetupWithoutHandshake.expectDATA(Http2ServerSpec.scala:968)
	at akka.http.impl.engine.http2.Http2ServerSpec$$anon$34.<init>(Http2ServerSpec.scala:738)
	at akka.http.impl.engine.http2.Http2ServerSpec.$anonfun$new$47(Http2ServerSpec.scala:714)
	at akka.stream.testkit.scaladsl.StreamTestKit$.assertAllStagesStopped(StreamTestKit.scala:32)
```

**Comment by jrudolph** [comment]
https://jenkins.akka.io:8498/job/akka-http-nightly/AKKA_VERSION=master,Node=akka-http-nightly-node,SCALA_VERSION=2.12.10,jdk=AdoptOpenJDK%208/771/testReport/junit/akka.http.impl.engine.http2/Http2ServerSpec/The_Http_2_server_implementation_should_support_stream_support_for_sending_response_entity_data_should_handle_RST_STREAM_while_waiting_for_a_window_update/

**Comment by jrudolph** [comment]
Actually the second one is a different case. Opened https://github.com/akka/akka-http/issues/2907 instead.

**Comment by raboof** [comment]
https://jenkins.akka.io:8498/job/akka-http-nightly/AKKA_VERSION=master,Node=akka-http-nightly-node,SCALA_VERSION=2.13.2,jdk=AdoptOpenJDK%208/177/consoleFull

Different line but same test and same message:
```
[info]   - should close substreams when connection is shutting down *** FAILED ***
[info]     WINDOW_UPDATE was not equal to DATA (Http2FrameProbe.scala:177)
```

**Comment by ennru** [comment]
Should be fixed by #3314
